### PR TITLE
Fix query template issues for bulk load

### LINKED ISSUE #42061: MySqlHook bulk_load_custom() syntax errors due to escaped sql command
### Apache Airflow Provider(s)

mysql

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon[aiobotocore]==8.17.0
apache-airflow-providers-apprise==1.2.1
apache-airflow-providers-atlassian-jira==1.1.0
apache-airflow-providers-celery==3.5.2
apache-airflow-providers-common-sql==1.10.1
apache-airflow-providers-ftp==3.7.0
apache-airflow-providers-google==10.14.0
apache-airflow-providers-http==4.9.0
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-microsoft-mssql==3.6.0
apache-airflow-providers-mysql==5.5.2
apache-airflow-providers-postgres==5.10.0
apache-airflow-providers-sftp==4.8.1
apache-airflow-providers-snowflake==5.3.0
apache-airflow-providers-sqlite==3.7.0
apache-airflow-providers-ssh==3.10.0
apache-airflow-providers-tableau==4.4.1

### Apache Airflow version

2.7.2

### Operating System

Amazon Linux 2023

### Deployment

Amazon (AWS) MWAA

### Deployment details

Python 3.11

### What happened

Using the below task always fails due to a MySQL syntax error as the "IGNORE" Duplicate-Key and Error Handling option and any mysql_extra_options values are escaped before execution.
```
load_data= S3ToMySqlOperator(
  task_id="load_data",
  s3_source_key=f"s3://{S3_BRONZE_BUCKET}/test.csv",
  mysql_table="test_db.test_table",
  mysql_conn_id=MYSQL_CONN_ID,
  mysql_extra_options="FIELDS TERMINATED BY ',' (col1, col3)",
  mysql_local_infile=True,
)
```

### What you think should happen instead

The task instance fails with the following error:
`MySQLdb.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''IGNORE' INTO TABLE test_db.test_table 'FIELDS TERMINATED BY \\',\\' (col1,' at line 1")`

### How to reproduce

In a MySQL environment with local-infile enabled:
```
CREATE TABLE test_db.test_table (
	col1 INT NULL,
	col2 INT NULL,
	col3 INT NULL
);
```

test.csv (create and upload to "test_bucket"):
```
100,200
300,400
500,600
```

example_dag.py
```
"""An example dag."""

from __future__ import annotations

from pathlib import Path

import pendulum

from airflow import DAG
from airflow.providers.common.sql.operators.sql import SQLExecuteQueryOperator
from airflow.providers.mysql.transfers.s3_to_mysql import S3ToMySqlOperator

MYSQL_CONN_ID = "mysql_server"
S3_BUCKET = "test_bucket"

with DAG(
    dag_id=Path(__file__).name.replace(".py", ""),
    start_date=pendulum.datetime(2022, 1, 1, tz="Europe/London"),
    schedule=None,
):
    truncate_test_table = SQLExecuteQueryOperator(
        task_id="truncate_test_table",
        sql="TRUNCATE TABLE test_db.test_table",
        autocommit=True,
        conn_id=MYSQL_CONN_ID,
        database="wallboard",
    )

    load_data= S3ToMySqlOperator(
        task_id="load_data",
        s3_source_key=f"s3://{S3_BUCKET}/test.csv",
        mysql_table="test_db.test_table",
        mysql_conn_id=MYSQL_CONN_ID,
        mysql_extra_options="FIELDS TERMINATED BY ',' (col1, col3)",
        mysql_local_infile=True,
    )

    truncate_test_table >> load_data
```

Running the DAG should result in an error of
`MySQLdb.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''IGNORE' INTO TABLE test_db.test_table 'FIELDS TERMINATED BY \\',\\' (col1,' at line 1")`

### Anything else

Occurs every time because IGNORE is passed as default and gets escaped incorrectly.

I believe it was caused by this PR https://github.com/apache/airflow/pull/33328 which added the escapes to prevent sql injection but seems to have broken the bulk_load_custom() method,

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by michaelimas1**
closes: #42061

Refactored SQL command to interpolate duplicate_key_handling and extra_options directly, preventing unwanted quotes around strings.

Since [this](https://github.com/apache/airflow/commit/3598a52a2d1a7f9aef5e3777f1dd2a5823722aa2) change, custom bulk load in mysql is broken. that change made the strings to be inserted to the query with quotes which resulted SQL error from mysql.

`(1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''REPLACE' INTO TABLE my_table ''' at line 1");`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — COMMENTED** [review]
Could you add a unit test for this change?

**Comment by michaelimas1** [comment]
Done

File: tests/providers/mysql/hooks/test_mysql.py [hunk 1]
```diff
@@ -290,25 +290,20 @@ def test_bulk_dump(self):
 
     def test_serialize_cell(self):
         assert "foo" == self.db_hook._serialize_cell("foo", None)
-
+
```
Comment (eladkal) [inline tests/providers/mysql/hooks/test_mysql.py]:
```suggestion

```

**Comment by michaelimas1** [comment]
@eladkal anything else missing from my side to get this approved? it's my first PR here so i'm not sure what's the process

**Review by potiuk — CHANGES_REQUESTED** [review]
This one introduces a security issue (one that https://github.com/apache/airflow/pull/33328 attempted to prevent). 

We absolutely cannot build an SQL query with interpolated value that can be provided from outside - this opens up all kind of possible manipulations with the query that might lead to a number of security issues. 

The only way we can accept such a change if it actually checks allowed values for duplicate_key_handling and raises an exception it's not strictly one of the two values expected.

**Comment by potiuk** [comment]
Same with extra-options. This is classic [SQL Injection](https://en.wikipedia.org/wiki/SQL_injection) and you have to find a way to sanitize the input in the way that it cannot be abused .

**Pull request closed without merging**